{
    "id": "LUCENE-2307",
    "title": "Spurious exception in TestIndexWriter",
    "details": {
        "labels": "",
        "priority": "Major",
        "components": [
            "core/index"
        ],
        "type": "Bug",
        "fix_versions": [],
        "affect_versions": "3.0.2",
        "resolution": "Cannot Reproduce",
        "status": "Closed"
    },
    "description": "Happened on trunk:\n\n    [junit] Testsuite: org.apache.lucene.index.TestIndexWriter\n    [junit] Tests run: 106, Failures: 1, Errors: 0, Time elapsed: 18.567 sec\n    [junit] \n    [junit] ------------- Standard Output ---------------\n    [junit] java.io.IOException: background merge hit exception: _0:c19 _1:c19 _2:c2 _3:c2 _4:c2 _5:c2 _6:c2 _7:c2 _8:c1 into _9 [optimize] [mergeDocStores]\n    [junit] \tat org.apache.lucene.index.IndexWriter.optimize(IndexWriter.java:2341)\n    [junit] \tat org.apache.lucene.index.IndexWriter.optimize(IndexWriter.java:2280)\n    [junit] \tat org.apache.lucene.index.IndexWriter.optimize(IndexWriter.java:2250)\n    [junit] \tat org.apache.lucene.index.TestIndexWriter$CommitAndAddIndexes.doBody(TestIndexWriter.java:3759)\n    [junit] \tat org.apache.lucene.index.TestIndexWriter$RunAddIndexesThreads$1.run(TestIndexWriter.java:3708)\n    [junit] Caused by: org.apache.lucene.index.MergePolicy$MergeAbortedException: merge is aborted: _0:c19 _1:c19 _2:c2 _3:c2 _4:c2 _5:c2 _6:c2 _7:c2 _8:c1 into _9 [optimize] [mergeDocStores]\n    [junit] \tat org.apache.lucene.index.MergePolicy$OneMerge.checkAborted(MergePolicy.java:135)\n    [junit] \tat org.apache.lucene.index.IndexWriter.mergeMiddle(IndexWriter.java:4308)\n    [junit] \tat org.apache.lucene.index.IndexWriter.merge(IndexWriter.java:3958)\n    [junit] \tat org.apache.lucene.index.ConcurrentMergeScheduler.doMerge(ConcurrentMergeScheduler.java:339)\n    [junit] \tat org.apache.lucene.index.ConcurrentMergeScheduler$MergeThread.run(ConcurrentMergeScheduler.java:407)\n    [junit] ------------- ---------------- ---------------\n    [junit] Testcase: testAddIndexesWithClose(org.apache.lucene.index.TestIndexWriter):\tFAILED\n    [junit] null\n    [junit] junit.framework.AssertionFailedError: null\n    [junit] \tat org.apache.lucene.index.TestIndexWriter.testAddIndexesWithClose(TestIndexWriter.java:3835)\n    [junit] \tat org.apache.lucene.util.LuceneTestCase.runBare(LuceneTestCase.java:252)\n    [junit]",
    "attachments": {},
    "issue_links": {},
    "comments": [
        {
            "date": "2010-03-11T10:38:31+0000",
            "content": "This is odd \u2013 that test calls close(true) on IW, meaning wait for all merges to complete.  So offhand I don't know why a merge is being aborted... that should only happen if close(false) is called.  Strange.... ",
            "author": "Michael McCandless",
            "id": "comment-12844004"
        },
        {
            "date": "2010-12-02T23:47:50+0000",
            "content": "Never reproduced since, closing as stale and outdated ",
            "author": "Earwin Burrfoot",
            "id": "comment-12966351"
        }
    ]
}